packages/telegram/src/getTelegramChatHistory.ts

```diff
@@ -0,0 +1,46 @@
+import { prisma } from '@charmverse/core/prisma-client';
+
+/**
+ * Return the chat history for a given Telegram chat ID, always in chronological order
+ * @limit - The number of messages to return, starting from the most recent
+ */
+export async function getTelegramChatHistory({
+  telegramChatId,
+  maxAgeInMinutes,
+  limit
+}: {
+  telegramChatId: string | number;
+  maxAgeInMinutes?: number;
+  limit?: number;
+}) {
+  if (!telegramChatId) {
+    throw new Error('Telegram chat ID is required');
+  }
+
+  let skip = 0;
+
+  if (limit) {
+    const totalMessages = await prisma.agentTelegramMessage.count({
+      where: {
+        conversationTelegramId: Number(telegramChatId)
+      }
+    });
+    // If the total number of messages is less than the limit, set skip to 0
+    skip = Math.max(0, totalMessages - Number(limit));
```

Comment:
its not clear why we would ever skip messages. The comment is confusing too since skip is 0 by default.

Reply:
Did AI write this? I think that this functino does give you the most recent comments , but it is out of order since it is giving oldest first. You should just order by date descending and pass the limit. No need to skip anything

Reply:
There is a mistake in the description.

It is purposefully in ascending order since we want the consumer of the method to be able to pass the messages as is to an AI, without additional sorting.

Reply:
The reason we want to skip old messages is for context window trimming. We don't want to always pass the full history (especially with cheaper models)

This method should remain unopiniated about the max context size, but provide controls to the call to define how many recent messages they want